generated-sample-code/samplegen_basics.js

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// DO NOT EDIT! This is a generated sample ("Request",  "samplegen_basics")
+
+// sample-metadata:
+//   title: This is the sample title
+//   description: This is the sample description
+//   usage: node samples/v1/samplegen_basics.js [--display_name "This is the default value of the display_name request field"]
+
+'use strict';
+
+function main(displayName = 'This is the default value of the display_name request field') {
+  // [START samplegen_basics]
+  /**
+  * TODO(developer): Uncomment these variables before running the sample.
+  */
+  // const displayName = 'This is the default value of the display_name request field';
+
+  // Imports the client library
+  const {ProductSearchClient} = require('@google-cloud/vision').v1;
+
+  // Instantiates a client
+  const productSearchClient = new ProductSearchClient();
+
+  async function sampleCreateProductSet() {
```

Comment:
Can we pass a client to this function as a parameter? It will highlight the idea that the client is created once and can be reused. Otherwise, folks will just erase the `function sampleCreateProductSet()` line and will get a sample that instantiates and uses the client, and will copy-paste it creating multiple clients, which is not what we want.

Reply:
**Note:** this syntax taken directly from [Node.js canonical sample examples](https://g3doc.corp.google.com/company/teams/cloud-devrel/dpe/samples/nodejs.md?cl=head#canonical-samples) in [code sample standards guide](http://go/cloud-samples)